modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/ScanVersionsCursor.java

```diff
@@ -63,34 +63,42 @@ public void close() {
 
     @Override
     public boolean hasNext() {
-        if (hasNext != null) {
-            return hasNext;
-        }
+        return storage.busy(() -> {
```

Comment:
Not long ago I heard that `busyLock` cause performance degradation.
Maybe we shouldn't use them in the cursor?
Maybe it's enough to check some field?

Reply:
I think we must first guarantee correctness and safety and only then optimize. Here, I just added a missing guard, `busy()` is still taken by other cursors.

If there is actually a problem with the current approach, we should probably change it for all cursors, it's a subject for another task.